UnusedImports: value javadoc tag does not reference imports

Javadoc `{@value}` tag can reference static constants and makes constant value appear in the generated javadoc. Constant can be referenced either by fully qualified name `{@value java.util.Calendar#MONTH}` or by relative name `{@value Calendar#MONTH}` when corresponding import is present `import java.util.Calendar;`.

```
$ javac InputUnusedImportWithValueTag.java
$ cat InputUnusedImportWithValueTag.java
import java.util.Calendar; // import is used in javadoc {@value}

public class InputUnusedImportWithValueTag {

    /**
     * Method determines current month as for {@value Calendar#MONTH}.
     *
     * @return index of the current month.
     */
    public int currentMonth() {
        return 1;
    }
}
$ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="UnusedImports">
      <property name="processJavadoc" value="true"/>
    </module>
  </module>
</module>
$ java -jar checkstyle-6.18-all.jar -c config.xml InputUnusedImportWithValueTag.java
Starting audit...
[ERROR] InputUnusedImportWithValueTag.java:1:8: Unused import - java.util.Calendar. [UnusedImports]
Audit done.
Checkstyle ends with 1 errors.
```
I'd be happy to contribute and fix this issue. Tentative fix: https://github.com/lutovich/checkstyle/commit/25c5b4e5060d8de22d0d12a121960e69d4bdf746.
Please let me know if my help is needed or PR should be created from this branch.

Thanks!

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTag.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTag.java
@@ -142,6 +142,7 @@ public class JavadocTag {
     public boolean canReferenceImports() {
         return tagInfo == JavadocTagInfo.SEE
                 || tagInfo == JavadocTagInfo.LINK
+                || tagInfo == JavadocTagInfo.VALUE
                 || tagInfo == JavadocTagInfo.LINKPLAIN
                 || tagInfo == JavadocTagInfo.THROWS
                 || tagInfo == JavadocTagInfo.EXCEPTION;

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheckTest.java
@@ -105,6 +105,13 @@ public class UnusedImportsCheckTest extends BaseCheckTestSupport {
     }
 
     @Test
+    public void testProcessJavadocWithLinkTag() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(UnusedImportsCheck.class);
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputUnusedImportWithValueTag.java"), expected);
+    }
+
+    @Test
     public void testAnnotations() throws Exception {
         final DefaultConfiguration checkConfig = createCheckConfig(UnusedImportsCheck.class);
         final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/InputUnusedImportWithValueTag.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/InputUnusedImportWithValueTag.java
@@ -1,0 +1,15 @@
+package com.puppycrawl.tools.checkstyle.checks.imports;
+
+import java.util.Calendar;
+
+public class InputUnusedImportWithValueTag {
+
+    /**
+     * Method determines current month as for {@value Calendar#MONTH}.
+     *
+     * @return index of the current month.
+     */
+    public int currentMonth() {
+        return 1;
+    }
+}

